docs/user_guide/building_blocks/config_objects.md

```diff
@@ -53,9 +125,374 @@ given via the ```--variables``` argument, accessing information from columns bas
+a way to create the list of LFNs (= the paths to the root files starting from `/store`) corresponding to this dataset and
+return this list. Two other auxiliary parameters can be changed, these are called
+`cfg.x.get_dataset_lfns_sandbox` and `cfg.x.get_dataset_lfns_remote_fs`. `get_dataset_lfns_sandbox`
+provides the sandbox in which the task GetDatasetLFNS will be run and expects therefore a
+{external+law:py:class}`law.sandbox.base.Sandbox` object, which can be for example obtained through
+the {py:func}`~columnflow.util.dev_sandbox` function.
+`get_dataset_lfns_remote_fs` provides the remote file system on which the LFNs for the specific dataset can be found, it expects a function with the `dataset_inst` as a parameter and returning the name of the file system as defined in the law config file.
+
+An example of such a function and the definition of the corresponding config parameters for a campaign where all datasets have been custom processed and stored on a single remote file system is given below.
+
+```python
+# custom lfn retrieval method in case the underlying campaign is custom uhh
+if cfg.campaign.x("custom", {}).get("creator") == "uhh":
+    def get_dataset_lfns(
+        dataset_inst: od.Dataset,
+        shift_inst: od.Shift,
+        dataset_key: str,
+    ) -> list[str]:
+        # destructure dataset_key into parts and create the lfn base directory
+        dataset_id, full_campaign, tier = dataset_key.split("/")[1:]
+        main_campaign, sub_campaign = full_campaign.split("-", 1)
+        lfn_base = law.wlcg.WLCGDirectoryTarget(
+            f"/store/{dataset_inst.data_source}/{main_campaign}/{dataset_id}/{tier}/{sub_campaign}/0",
+            fs=f"wlcg_fs_{cfg.campaign.x.custom['name']}",
+        )
+
+        # loop though files and interpret paths as lfns
+        return [
+            lfn_base.child(basename, type="f").path
+            for basename in lfn_base.listdir(pattern="*.root")
+        ]
+
+    # define the lfn retrieval function
+    cfg.x.get_dataset_lfns = get_dataset_lfns
+
+    # define a custom sandbox
+    cfg.x.get_dataset_lfns_sandbox = dev_sandbox("bash::$CF_BASE/sandboxes/cf.sh")
+
+    # define custom remote fs's to look at
+    cfg.x.get_dataset_lfns_remote_fs = lambda dataset_inst: f"wlcg_fs_{cfg.campaign.x.custom['name']}"
+```
+
+### External_files
+
+If some files from outside columnflow are needed for an analysis, be them local files or online ( and accessible through wget), these can be indicated in the `cfg.x.external_files` auxiliary parameter.
+These can then be copied to the columnflow outputs using the {py:class}`~columnflow.tasks.external.BundleExternalFiles` task and used by being required by the object needing them.
+The `cfg.x.external_files` parameter expects a {py:class}`~columnflow.util.DotDict` with a string
+explaining the role of the file to be downloaded/copied as key and the link/path as value.
```

Comment:
> ... explaining the role of the file to be downloaded/copied as key ...

What do you mean by this? There is no requirement/naming convention at all for these keys, right? If the user so wishes, they could in principle use any kind of key

Reply:
yes, this would be a best pratices case, I reformulated